File: games/tbamud/commands/movement.py
"""Movement commands — extended door interactions, enter/leave."""

from __future__ import annotations

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from core.engine import Engine
    from core.session import Session


def register(engine: Engine) -> None:
    engine.register_command("enter", do_enter, korean="들어가")
    engine.register_command("leave", do_leave, korean="나와")
    engine.register_command("follow", do_follow, korean="따라가")
    engine.register_command("group", do_group, korean="무리")


async def do_enter(session: Session, args: str) -> None:
    if not args:
        await session.send_line("어디로 들어가시겠습니까?")
        return
    await session.send_line("구현 예정입니다.")


async def do_leave(session: Session, args: str) -> None:
    await session.send_line("구현 예정입니다.")


async def do_follow(session: Session, args: str) -> None:
    if not args:
        await session.send_line("누구를 따라가시겠습니까?")
        return
    await session.send_line("구현 예정입니다.")


async def do_group(session: Session, args: str) -> None:
    if not args:
        await session.send_line("파티원이 없습니다.")
        return
    await session.send_line("구현 예정입니다.")

File: games/tbamud/lua/commands/movement.lua
-- movement.lua — tbaMUD extended movement commands

register_command("enter", function(ctx, args)
    if not args or args == "" then
        ctx:send("어디로 들어가시겠습니까?")
        return
    end
    ctx:send("구현 예정입니다.")
end, "들어가")

register_command("leave", function(ctx, args)
    ctx:send("구현 예정입니다.")
end, "나와")

register_command("follow", function(ctx, args)
    if not args or args == "" then
        ctx:send("누구를 따라가시겠습니까?")
        return
    end
    ctx:send("구현 예정입니다.")
end, "따라가")

register_command("group", function(ctx, args)
    if not args or args == "" then
        ctx:send("파티원이 없습니다.")
        return
    end
    ctx:send("구현 예정입니다.")
end, "무리")
